polynote-frontend/polynote/data/data.ts

```diff
@@ -71,7 +71,15 @@ export abstract class RepositoryConfig extends CodecContainer {
     static codecs: typeof RepositoryConfig[];
     static msgTypeId: number;
 
-    abstract url: string
+    abstract url: string;
+    abstract repositoryTypeName: "ivy" | "maven" | "pip";
```

Comment:
[nit] if you define a [type alias](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases) you only need to type `"ivy" | "maven" | "pip"` once (and it makes it a little easier to add a resolver later, not like that its terribly likely to happen tho)